### PR TITLE
Default language, add exception handler, prepare for shortend search results

### PR DESCRIPTION
**Pull request by joeimer**

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -61,16 +63,20 @@ async def answer(message, embed):
     except discord.errors.Forbidden:
         log.warning(
             f'[{message.guild.name}][{message.channel}] Could not post response, channel is forbidden for me.')
+    except discord.errors.HTTPException:
+        log.warning(
+            f'[{message.guild.name}][{message.channel}] Could not post response, message exceeds maximum size of 6000.')
```
Comment (maduck) [inline bot.py:68]: oh that's broad. Catching all HTTP Exceptions, but giving size limit error, without exception text?

File: bot.py [hunk 2]
```diff
@@ -756,6 +780,49 @@ def trim_content_to_length(text, link, max_length=1000):
...
+    async def change_language(self, message, new_language, prefix, lang):
+        my_language = self.language.get(message.guild)
+        if not message.guild:
+            e = discord.Embed(title='Language change', color=self.RED)
+            e.add_field(name='Error',
+                        value=f'Language change not possible in direct messages.')
+            await answer(message, e)
+            return
+        if self.is_guild_admin(message):
+            if len(new_language) != 2:
```
Comment (maduck) [inline bot.py:792]: i would remove that, since next check is for literal whitelisting.

File: language.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            with open(self.LANGUAGE_CONFIG_FILE) as f:
+                self.__languages = json.load(f)
+
+    def save_languages(self):
+        lock = threading.Lock()
+        with lock:
+            with open(self.LANGUAGE_CONFIG_FILE, 'w') as f:
+                json.dump(self.__languages, f, sort_keys=True, indent=2)
+
+    def add(self, guild, language):
```
Comment (maduck) [inline language.py:28]: that should be "set" rather than "add", because it does replacing action and not appending.

File: language.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        self.__languages[str(guild.id)] = language
```
Comment (maduck) [inline language.py:29]: in the others I created a dictionary with the guild's name also, to be able to give support. The guild id is fine for storage, but not human readable.

File: bot.py [hunk 5]
```diff
@@ -756,6 +780,49 @@ def trim_content_to_length(text, link, max_length=1000):
...
+            if new_language not in Translations.LANGUAGES:
+                e = discord.Embed(title='Language change', color=self.RED)
+                e.add_field(name='Error',
+                            value=f'Your new language is not available, `{new_language}`.')
+                await answer(message, e)
+                return
+
+            self.language.add(message.guild, new_language)
+            e = discord.Embed(title='Administrative action', color=self.RED)
+            e.add_field(name='Language change', value=f'Language was changed from `{my_language}` to `{new_language}`')
```
Comment (maduck) [inline bot.py:807]: The scope of the change is not clear. Let's make it easy to understand that it's default for the whole server now.

File: language.py (hunk 4 above)
Comment (maduck) [reply language.py:29]: Also, you copied my shitty `str(guild_id)` ... please remove the type cast here, it does not make sense.

File: language.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        self.save_languages()
+
+    def get(self, guild):
+        if guild is None:
+            return self.default_language
+        return self.__languages.get(str(guild.id), self.default_language)
```
Comment (maduck) [inline language.py:35]: same here

**Comment by joeimer** [comment]
I just copied and paste the structure of the prefix function so it looks the same. I will close the pull request and think about it later the days,

File: bot.py (hunk 1 above)
Comment (joeimer) [reply bot.py]: I removed the handler. Let's add a handler for this later

**Pull request merged**